src/pages/patientView/mutation/PatientViewMutationTable.tsx

```diff
@@ -33,6 +34,7 @@ export interface IPatientViewMutationTableProps extends IMutationTableProps {
     onFilterGenes?: (option: GeneFilterOption) => void;
     onSelectGenePanel?: (name: string) => void;
     disableTooltip?: boolean;
+    existsSomeMutationWithAscnProperty: { [property: string]: boolean };
```

Comment:
Should this just be an optional if the ascn is not available?

Reply:
We can make that change if you are requesting it.

This map / dictionary will determine whether certain columns are visible in the patient view and in the results view. In those locations this map is consulted to see whether ascn-specific mutation fields are present and populated or not. This is a computed value based on the .result of an api remoteData promise. When that promise has not yet been satisfied, the computed value will have all ascn fields mapped to false. When the promise is fulfilled, the data is examined and whichever fields are populated in some mutation -- those fields get mapped to true. We hide or show particular columns based on whether individual fields have a value set in some mutation. For example, if there is a mutation with the ascn field "ccfMCopies" populated, then the "Clonal" column will be made visible. Other columns may not be visible .. or instance if there is no mutation with a value set for the field ascnMethod then the ASCN Method column will not be visible. Because we make the decision for showing facets fields on a field-by-field basis, this computed map shows whether we know a mutation exists with a value for each field.

Currently, when the remoteData promise is unfulfilled, we return a populated map with "false" for each field of interest. If you would like, we can return 'undefined' when the promise is unfulfilled and then add logic in the columnMappers to say that the columns should be excluded when the map is either undefined or where the map contains a 'false' in the needed columns. If you want that, just let us know.